File: labs/G60_Composite/python/startingpoint/position.py
from datetime import datetime
from enum import Enum


class Direction(Enum):
    Long = 0
    Short = 1


class Status(Enum):
    Open = 0
    Closed = 1


Symbol = str


class Position:
    symbol: Symbol
    quantity: int
    opened_on: datetime
    closed_on: datetime
    direction: Direction
    status: Status

    def getInfo(self) -> str:
        return '[' + self.symbol + ' ' + str(self.quantity) + ' ' + self.direction.name + ']'

    def __init__(self, symbol: Symbol, quantity: int, direction: Direction=Direction.Long):
        self.symbol=symbol
        self.quantity=quantity
        self.direction=direction
        self.opened_on=datetime.utcnow()
        self.status=Status.Open

File: labs/G60_Composite/java/startingpoint/app/src/main/java/CompositeStartingPoint/Position.java
package CompositeStartingPoint;

import java.time.LocalDateTime;

public class Position {
    private LocalDateTime openedOn;
    private LocalDateTime closedOn;
    private Direction direction;
    private String symbol;
    private int quantity;

    public void close() throws Exception {
        if(this.closedOn!=null){
            throw new Exception("Already closed!");
        }

        this.closedOn=LocalDateTime.now();
    }

    public LocalDateTime getOpenedOn() {
        return openedOn;
    }

    public LocalDateTime getClosedOn() {
        return closedOn;
    }

    public Direction getDirection() {
        return direction;
    }

    public String getSymbol() {
        return symbol;
    }

    public int getQuantity() {
        return quantity;
    }


    @Override
    public String toString() {
        return "Position [closedOn=" + closedOn + ", direction=" + direction + ", openedOn=" + openedOn + ", quantity="
                + quantity + ", symbol=" + symbol + "]";
    }

    public static Position Open(String symbol, int quantity,Direction direction) {
        Position p = new Position();

        p.direction = direction;        
        p.symbol = symbol;
        p.quantity = quantity;
        p.openedOn=LocalDateTime.now();

        return p;
    }


}
